lib/bambora/v1/payment_resource.rb

```diff
@@ -77,6 +77,22 @@ def create_with_payment_profile(customer_code:, amount:, card_id: 1, complete: f
       end
 
       alias make_payment_with_payment_profile create_with_payment_profile
+
+      # Retrieve the details of a previously attempted payment.
+      #
+      #
+      # @example
+      #
+      #   client = Bambora::Rest::JSONClient(base_url: '...', api_key: '...', merchant_id: '...')
+      #   payments = Bambora::V1::PaymentResource(client: client)
+      #   payments.get(transaction_id: 1000341)
+      #
+      # @param transaction_id [Integer] An integer identifier for the associated transaction.
+      #
+      # @return [Hash] Transaction details.
+      def get(transaction_id:)
+        client.get(path: "#{sub_path}/#{transaction_id}", api_key: api_key)
+      end
```

Comment:
@harrylewis Does `client.get` also handle HTTP errors? What happens in case this returns a 401 or 502 for example?

Reply:
@cristianoventura good question. The `client` is an inherited instance of `Bambora::Rest::Client`, which has a method `#parse_response_body` shown below.

https://github.com/HiMamaInc/bambora-client/blob/e057af28cc490e3a49269f79206ee71cf4715deb/lib/bambora/rest/client.rb#L62-L64

You can see this calls on an adapter, which eventually delegates to an inherited instance of `Bambora::Response`, where `#error_response` is defined.

https://github.com/HiMamaInc/bambora-client/blob/e057af28cc490e3a49269f79206ee71cf4715deb/lib/bambora/adapters/response.rb#L36-L38

No `raise` is happening. Rather, a response is returned from `Faraday`, the dependency we use for networking, and eventually we will return a response to the caller to handle.

Reply:
Thanks for clarifying! 👍 🙌